### DISCUSSION TITLE
Looking for a bigger example of Rematch 2.

### DISCUSSION THREAD

**Discussion by IKoshelev** [rematch/rematch#885] [Q&A]
I'm looking for an example of code organization with Rematch 2 that goes beyond the trivial cases like counter or todos (something with at least a dozen entities involved). Are there any tutorials or repos I could study?

**Comment by semoal** [comment]
You can check the example all-plugins-react-ts, which is a bit bigger. 
I've used on my previous company Rematch with a large scale codebase and worked nicely.

Right now we're writing an official Rematch book, and we're building an Amazon like application, so when it's released I can notify you ;)

**Reply by IKoshelev** [reply]
Looking forward to the book and thank you for the example. 
Example provides most of what I need, but so far I haven't been able to find how to use `selectors` with hooks, neither in examples nor in documentation. Could you please advice? 

P.S. it's also strange that selectors require a plugin. Derived state is a must have in any application, IMHO, they fully belong in the core.

**Reply by semoal** [reply]
Selectors since the creation of Rematch 4 years ago always has been a plugin that's why we kept that like this, nowadays probably selectors had more sense like a core part, but as Rematch it's pretty easy to include plugins we decided to keep the `core` as much as lightweight as we could, and then extend functionality with plugins, in that way users with Redux can benefit in less than 1.7kb the power of Rematch.

About your main question, you're so totally agree that we didn't document that (I've never developed any project with Rematch selectors + hooks), and you're totally right that feature "isn't supported" by Rematch.

React-Redux library offers a way of using the `useSelector` hook to pass the `reselect` selectors:
https://react-redux.js.org/api/hooks#using-memoizing-selectors

But right now we can't reuse the Rematch selectors with this useSelector hook, give me a days to think and iteratee over this and I'll put this on my highest priority list.

Thanks so much for your interest =)

**Reply by IKoshelev** [reply]
I ended up going with `reselect` selectors for the most part. I also made a sample application, with a set of requirements representative of enterprise apps: [demo](https://uiappashowcases.blob.core.windows.net/$web/rematch2-advanced/index.html) . I must say, I REALLY like resulting implementation, to the point of Rematch2 becoming my go-to library for new apps.

Requirements and code are available [on my github](https://github.com/IKoshelev/ui-app-architecture-showcases), I was wondering, if this implementation canbe shared as an example of enterprise-grade Rematch2 app?
